proto/schedule/utils.go

```diff
@@ -0,0 +1,94 @@
+package schedule
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"io/ioutil"
+	"time"
+)
+
+const (
+	DateFormat              = "Mon 02 Jan 2006"
+	defaultStopTimeDays 	= 7
+	dayDuration             = 24 * time.Hour
+	defaultStopTimeDuration = defaultStopTimeDays * dayDuration
+)
+
+func (s *Schedule) Validate() error {
+	if s == nil {
+		return fmt.Errorf("nil Schedule")
+	}
+
+	if s.Shifts == nil || len(s.Shifts) == 0 {
+		fmt.Printf("No input shifts found. Starting from scratch.")
+		return nil
+	}
+
+	for i, shift := range s.Shifts {
+		if _, err := time.Parse(DateFormat, shift.StartDate); err != nil {
+			return fmt.Errorf("error in input shift entry %v, invalid value: %v, err: %v", i, shift.StartDate, err)
+		}
+		if shift.User == "" {
+			return fmt.Errorf("user cannot be empty in input shift entry %v", i)
+		}
+	}
+
+	return nil // It's all good.
+}
+
+func FromYAMLFile(path string) (*Schedule, error) {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	s := &Schedule{}
+	if err := yaml.Unmarshal(b, s); err != nil {
+		return nil, err
+	}
+	return s, nil
```

Comment:
Probably should call `s.Validate()` here to make sure we're giving back a valid `Schedule`?

Reply:
I think reading from a well formed proto is separate from whether the data contained within is valid. The only time we need to ensure it's validated is before making the destructive `Clear` and subsequent new `Insert` calls

Reply:
Do we want to extend the schedule if the schedule data is invalid? Seems like it would be good to blow up there, too... IMO it feels like trying to do anything with invalid (but perhaps well-formed) data is probably a bad idea...

Reply:
Ok, that makes sense. Added call to Validate after unmarhsalling